### PR TITLE
[Infineon][1.0] Add args for CYW30739 to provision DAC/PAI/CD.

### PR DESCRIPTION
**Pull request by hsusid**
#### Problem
It's not supported to flash non-test DAC/PAI/CD into a CYW30739 device for a production build.

#### Change overview
* Implement DeviceAttestationCredentialsProvider APIs by FactoryDataProvider for CYW30739.
* Add GN args to provision DAC/PAI/CD from specified der files into CYW30739 flash.
* Obsolete CYW30739 static_config.txt by GN args.
* CYW30739Config loads factory data only from SS flash section.

#### Testing
chip-tool pairing
OTA testing

### CODE REVIEW

File: examples/lighting-app/infineon/cyw30739/include/CHIPProjectConfig.h [hunk 1]
```diff
@@ -34,13 +34,13 @@
 #define CHIP_DEVICE_CONFIG_DEVICE_VENDOR_NAME "Infineon"
 
 /* The VendorID attribute of the Basic cluster. */
-#define CHIP_DEVICE_CONFIG_DEVICE_VENDOR_ID 0xFFF1
+#define CHIP_DEVICE_CONFIG_DEVICE_VENDOR_ID 0xFFF2
```
Comment (andy31415) [inline examples/lighting-app/infineon/cyw30739/include/CHIPProjectConfig.h:37]: Both of these are test IDs ... is this change needed? Does it have to do with DAC/PAI/CD?

File: third_party/infineon/cyw30739_sdk/scripts/gen_ss_cgs.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--ss_hdf", required=True, type=pathlib.Path)
+    parser.add_argument("--ss_cgs", required=True, type=pathlib.Path)
+    parser.add_argument("--empty_cgs", required=True, type=pathlib.Path)
+    return parser.parse_args()
+
+
+def add_thread_factory_config(configs: OrderedDict):
```
Comment (andy31415) [inline third_party/infineon/cyw30739_sdk/scripts/gen_ss_cgs.py:54]: why is the param in/out?

usually it is more pythonic to return, so you would have `def get_thread_factory_config() -> OrderedDict`

Comment (hsusid) [reply third_party/infineon/cyw30739_sdk/scripts/gen_ss_cgs.py]: fixed

File: third_party/infineon/cyw30739_sdk/scripts/gen_ss_cgs.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        cgs.write(indent + data[i:i+16].hex(' ') + "\n")
+
+    cgs.write("""\
+    } END_COMMENTED_BYTES
+}
+""")
+
+
+if __name__ == "__main__":
+    sys.exit(main())
```
[the remark this answers is not in the document]
Comment (hsusid) [reply third_party/infineon/cyw30739_sdk/scripts/gen_ss_cgs.py]: fixed

File: examples/lighting-app/infineon/cyw30739/include/CHIPProjectConfig.h (hunk 1 above)
Comment (hsusid) [reply examples/lighting-app/infineon/cyw30739/include/CHIPProjectConfig.h:37]: Yes, it's needed.
CYW30739 example apps use the test DAC with VID 0xFFF2 by default.
Test IDs have to be identical to IDs in the test DAC for chip-tool pairing.

**Comment by hsusid** [comment]
rebased

**Review by woody-apple — APPROVED** [review]

**Comment by hsusid** [comment]
rebased for resolving conflicting files

**Review by andy31415 — APPROVED** [review]
